From 8b0c3c3813dd7149ab7f791ae7a13d67eabfdedf Mon Sep 17 00:00:00 2001
From: Huanle Han <hanxueluo@gmail.com>
Date: Wed, 12 Jul 2017 22:35:58 +0800
Subject: [PATCH 1/1] lacp: enable bond slave immediately after lacp attach

There is a long interval (5~20 seconds) between lacp slave attach
and bond slave enable. During the interval, ovs drop all received
packets from that slave because bond_check_admissibility() check
fails. The root cause is that connectivity_seq is not changed
after lacp update and lacp status is not populated into port->may_enable
by port_run() immediately.

Signed-off-by: Huanle Han <hanxueluo@gmail.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
---
 lib/lacp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/lacp.c b/lib/lacp.c
index 7716387f5..e6ad7b934 100644
--- a/lib/lacp.c
+++ b/lib/lacp.c
@@ -536,6 +536,7 @@ lacp_run(struct lacp *lacp, lacp_send_pdu *send_pdu) OVS_EXCLUDED(mutex)
 
     if (lacp->update) {
         lacp_update_attached(lacp);
+        seq_change(connectivity_seq_get());
     }
 
     HMAP_FOR_EACH (slave, node, &lacp->slaves) {
-- 
2.17.2

